prusti-tests/tests/verify/pass/extern-spec/module-arg.rs

```diff
@@ -0,0 +1,77 @@
+use prusti_contracts::*;
+
+fn main() {
+    use module::inner::*;
+
+    fn _trait_test_1<T: Example>() {
+        assert!(T::example() == 42)
+    }
+
+    fn _trait_test_2<T: Advanced<U>, U: Copy>() {
+        prusti_assert!(T::example() === T::example())
+    }
+
+    assert!(free_1() == 1);
+    assert!(free_2() == 2);
+    
+    let s = Struct;
+    assert!(s.method() == 42);
+}
+
+#[extern_spec(module::inner)]
+trait Example {
+    #[ensures(result == 42)]
+    fn example() -> i32;
+}
+
+#[extern_spec(module::inner)]
+trait Advanced<#[generic] T> where T: Copy {
+    #[pure]
+    fn example() -> T;
+}
+
+#[extern_spec(module)]
+mod inner {
+    use crate::*;
```

Comment:
I wonder: do we really need to leave these imports in? Do we keep the `mod` structure around when rewriting this kind of extern spec? Or is this just to appease IDEs?

Reply:
`#[extern_spec]` [erases](https://github.com/viperproject/prusti-dev/blob/a4ac54bbb747607ca9ab5d24f0013202a2e6e171/prusti-contracts/prusti-contracts-proc-macros/src/lib.rs#L73-L77) whatever it's attached to when not building for prusti, so I don't think this helps IDEs. I believe this is actually necessary because we're currently still preserving the module structure rather than moving everything to the top level. Thus, the `prusti_contracts` attributes are not in scope, and if you do import them, this case still gives an error because it can't resolve `module`, which is also not in scope within `inner`.